src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for setting up memory cache related services in an <see cref="IServiceCollection" />.
+    /// </summary>
+    public static class MemoryCacheServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds memory caching services to the specified <see cref="IServiceCollection" />.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        public static void AddMemoryCache(this IServiceCollection services)
+        {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.AddOptions();
+            services.TryAdd(ServiceDescriptor.Transient<IDistributedCache, LocalCache>());
```

Comment:
Hmm I think I now get why it was called `AddCaching` - it adds both a memory cache and a distributed cache. As such, I actually think that either this new name is wrong, or that it shouldn't add both types of caches. I'm thinking right now that the name is what's fine, and that it shouldn't add both types of caches.

Perhaps is should be:
- `AddMemoryCache` to add *just* `IMemoryCache` --> `MemoryCache`. The doc comment for this is simple.
- `AddMemoryDistributedCache` to add *just* `IDistributedCache` --> `LocalCache`. The doc comment for this needs to say that it's not *really* distributed, but rather a local in-memory implementation.

cc @Tratcher @lodejard